parser_library/src/processing/instruction_sets/asm_processor.cpp

```diff
@@ -817,7 +818,35 @@ void asm_processor::process_START(rebuilt_statement stmt)
 
     hlasm_ctx.ord_ctx.set_available_location_counter_value(start_section_alignment, offset);
 }
+void asm_processor::process_END(rebuilt_statement stmt)
+{
+    const auto& label = stmt.label_ref();
 
+    if (!check(stmt, hlasm_ctx, checker_, *this))
+        return;
+    if (!(label.type == semantics::label_si_type::EMPTY || label.type == semantics::label_si_type::SEQ))
+    {
+        add_diagnostic(diagnostic_op::warning_A249_sequence_symbol_expected(stmt.label_ref().field_range));
+        return;
```

Comment:
The invalid label type (or operand value below) should not suppress the end_reached.
```suggestion
```